### ISSUE TITLE
How can I show popover from code?

### ISSUE THREAD

**Issue by ildarnm** [pleerock/ng2-popover#15]

**Comment by pleerock** [comment]
Something like this should work: 

```typescript
        <button #myPopover popover="Hello popover">
            click here
        </button>
```
And you can use `#myPopover.open()` and `#myPopover.close()` methods.

```typescript
@ViewChildren("myPopover")
myPopover: Popover;

ngOnInit() {
    this.myPopover.open();
}
```

**Comment by voffkany** [comment]
I believe something has changed recently, so this example does not work:
1) There are no open() and close() methods.
2) When I copy pasted popover-content from your online demo and tried to call  this.myPopover.show() since the open method is not available - nothing happens. No error logs and the popover is not shown.
3) Why do you recommend @ViewChildren vs @ViewChild?

Thank you

**Comment by pleerock** [comment]
yeah it should be a `@ViewChild`, it was a typo, sorry about that.
Try to use [show](https://github.com/pleerock/ng2-popover/blob/master/src/Popover.ts#L104) or [hide](https://github.com/pleerock/ng2-popover/blob/master/src/Popover.ts#L154) methods instead

**Comment by voffkany** [comment]
No problem, but as I mentioned in the previous post the show method does not work either.  

May I ask you a favor to update the demo to include a full example how to show it programmatically? Basically, I'd like to see the code which will activate popover-content at line 22 at app/app.component.html file.

Thanks

**Comment by cristianmtr** [comment]
this doesn't work for me either:

html:
```
<popover-content #myPopoverContent [closeOnClickOutside]="true">
	[...]
</popover-content>

<div class="col-md-12 text-center">

	<button id="myPopover" #myPopover [popover]="myPopoverContent">
[...]
         </button>
	
</div>
```

ts:
```
[...]
 @ViewChild('myPopover')
  myPopover: Popover;

  ngAfterViewInit(): void {
    console.log('ngAfterViewInit');
    this.myPopover.show();
  }
[...]
```

Also tried:
- .open();
- different ng lifecycle hooks
- to refer the myPopoverContent

**Comment by cristianmtr** [comment]
As a workaround, you can get this working by simulating a 'click' event on the button.

```ts
/**
   * required since popover comp is broken
   * https://github.com/pleerock/ng2-popover/issues/15
   */
  showPopover(): void {
    let event = new MouseEvent('click', { bubbles: true });
    this.renderer.invokeElementMethod(
      this.myPopoverButton.nativeElement, 'dispatchEvent', [event]);
  }
```

**Comment by guzmo** [comment]
Same problem, also had the same problem with your ng2-tooltip. Tried getting it with ViewChild and trigger show() but nothing works.

**Comment by philjones88** [comment]
Works for me, 2.0.0 RTM of Angular2

**Comment by sufian07** [comment]
I also can not find a way to show popover programmatically. Someone help please

**Comment by pleerock** [comment]
don't forget that your `@ViewChild` wont work if your popover in some ngIf or other places. This is angular issue, not popover problem.

**Comment by dalibor-sanzeru** [comment]
Dirty (I suppose only one) way to this is in current implementation is to take reference to element used for source of click event (and can be evented manually from component code as well -> e.g. click to button) with @ViewChild and manually trigger click() event in component which use popover.

// CODE IN HTML - you can click at btn or use BE code to click it manually from component.
`<button type="button"  #btnClickSource [popover]="myPopover">Click to see popover</button>`

//CODE IN YOUR COMPONENT
```
//1. Import this
import { Renderer, ElementRef } from "@angular/core";

//2. Enrich constructor for renderer to access HTML from component.
constructor(private renderer: Renderer) {
    }

//3. Access to btn for manual event click on it.
@ViewChild("btnClickSource") btnClickSource: ElementRef;

//4. Code which should 'programmaticaly' execute OPEN Popover 
private beCodeToShowPopover() {
        // Dirty manual work with UI stuff to attach click event directly to button
        let event = new MouseEvent("click", { bubbles: true });
        this.renderer.invokeElementMethod(this.btnClickSource.nativeElement, "dispatchEvent", [event]);
    }
```
